### ISSUE TITLE
Slow uniform update behavior?

### ISSUE THREAD

**Issue by almarklein** [pygfx/pygfx#257]
When working on the Gizmo, it looked like updating the material uniforms of all elements) was slow (nearing one second). But I think the object uniforms are updated about on each draw (because the gizmo's matrix changes). This is odd and should be investigated  🤔

**Comment by almarklein** [comment]
I looked into this. I confirmed that the uniform buffer of all 18 or so objects is updated on each draw, and this has no noticeable performance penalty.

I'm now pretty sure that this issue was because of changing the opacity. When the opacity changes from opaque to transparent (or the other way around) the pipeline is rebuild, because it needs a different approach to get correct blending. And this takes *way* longer than a uniform update. I think that at the time, I first reset opacities to a default and then determined their actual values, so the chance of invoking a few pipeline rebuilds was pretty big. 

Ideally, the prop should remember the old value and determine whether it needs a rebuild JustInTime ... maybe someday.
